### ISSUE TITLE
feat: replace LogisticRegression with ensemble model toggle (Random Forest / XGBoost) with cross-validation metrics display — listed as Hard difficulty Good First Issue in README

### ISSUE THREAD

**Issue by divyanshim27** [Arnav-Singh-5080/IPL-Win-Probability-Prediction#607]
## Summary

The README's "Good First Issues" table explicitly lists:

> *"Random Forest / XGBoost toggle — Hard"*
> *"Cross-validation metrics — Medium"*

The current model is a single `LogisticRegression(max_iter=1000)` with no
model comparison, no cross-validation, and no uncertainty quantification.
This issue proposes implementing a model selector with k-fold cross-validation
metrics displayed in the Streamlit UI.

## Why LogisticRegression Alone Is Insufficient

The win probability prediction task is inherently non-linear:
- CRR/RRR interactions, wicket acceleration curves, and death-over patterns
  are all non-linear decision boundaries
- LogisticRegression assumes linear separability between win/loss states
- Calibration degrades at extreme match states (10 runs needed, 1 wicket left)

Random Forest and XGBoost capture these non-linear patterns and provide
better-calibrated probabilities via Platt scaling or isotonic regression.

### 1. Model selector in `train.py`

```python
# train.py
import argparse

MODELS = {
    "logistic": LogisticRegression(max_iter=1000),
    "random_forest": RandomForestClassifier(n_estimators=100, random_state=42),
    "xgboost": XGBClassifier(n_estimators=100, use_label_encoder=False,
                              eval_metric="logloss", random_state=42),
}

parser = argparse.ArgumentParser()
parser.add_argument("--model", choices=MODELS.keys(), default="logistic")
args = parser.parse_args()

# k-fold cross-validation
from sklearn.model_selection import cross_validate
cv_results = cross_validate(
    pipeline, X, y, cv=5,
    scoring=["accuracy", "roc_auc", "neg_brier_score"]
)
```

### 2. Model toggle in Streamlit sidebar

```python
# app.py
model_choice = st.sidebar.selectbox(
    "Prediction Model",
    ["Logistic Regression", "Random Forest", "XGBoost"],
    help="Select model. XGBoost is more accurate but slower."
)

model_map = {
    "Logistic Regression": "models/pipe_logistic.pkl",
    "Random Forest": "models/pipe_rf.pkl",
    "XGBoost": "models/pipe_xgb.pkl",
}

with open(model_map[model_choice], "rb") as f:
    pipe = pickle.load(f)
```

### 3. Cross-validation metrics display

```python
# Metrics card in Streamlit
st.sidebar.markdown("### Model Performance (5-Fold CV)")
st.sidebar.metric("Accuracy", f"{cv_accuracy:.1%}")
st.sidebar.metric("ROC-AUC", f"{cv_roc_auc:.3f}")
st.sidebar.metric("Brier Score", f"{cv_brier:.4f}")
```

## Acceptance Criteria

- [ ] `train.py` accepts `--model` argument: `logistic`, `random_forest`,
      `xgboost`
- [ ] 3 model PKLs saved to `models/` directory (`.gitignore`d)
- [ ] Streamlit sidebar model selector toggle
- [ ] 5-fold cross-validation metrics (Accuracy, ROC-AUC, Brier Score)
      displayed per model in the sidebar
- [ ] `requirements.txt` updated with `xgboost`
- [ ] README's Model table updated with comparative accuracy metrics

**Labels:** `enhancement`, `ml`, `good-first-issue`, `level: advanced`

**Comment by divyanshim27** [comment]
Hey @Arnav-Singh-5080   ,
I have deeply gone through the project repo with all-proper local running and testing , and now my observations and analysis , I found this as major lagging/enhancement in the project repo .
Based on my issue understanding and my skill-set for solving this problem, I would want to contribute to the issue.
Please assign it to me .
